tools/http-test-helper/src/main/java/org/enso/shttp/HybridHTTPServer.java

```diff
@@ -0,0 +1,139 @@
+    sslServer = HttpsServer.create(sslAddress, 0);
+    sslServer.setExecutor(null);
+  }
+
+  private static class SimpleHttpsConfigurator extends HttpsConfigurator {
+    public SimpleHttpsConfigurator(SSLContext context) {
+      super(context);
+    }
+
+    @Override
+    public void configure(HttpsParameters params) {
+      SSLContext ctx = getSSLContext();
+      SSLEngine engine = ctx.createSSLEngine();
+      params.setNeedClientAuth(false);
+      params.setCipherSuites(engine.getEnabledCipherSuites());
+      params.setProtocols(engine.getEnabledProtocols());
+      params.setSSLParameters(ctx.getDefaultSSLParameters());
+    }
+  }
+
+  private void setupSSL() throws Exception {
+    String password = "test-password";
+    SSLContext context = SSLContext.getInstance("TLS");
+    KeyStore keyStore = initializeKeyStore(password);
+
+    KeyManagerFactory keyManagerFactory =
+        KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+    keyManagerFactory.init(keyStore, password.toCharArray());
+    TrustManagerFactory trustManagerFactory =
+        TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+    trustManagerFactory.init(keyStore);
+
+    context.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+    sslServer.setHttpsConfigurator(new SimpleHttpsConfigurator(context));
+  }
+
+  private KeyStore initializeKeyStore(String password) throws Exception {
+    KeyStore keyStore = KeyStore.getInstance("JKS");
+
+    if (Files.exists(keyStorePath)) {
+      Files.delete(keyStorePath);
+    }
+
+    int exitCode =
+        (new ProcessBuilder())
+            .command(
+                "keytool",
+                "-genkey",
+                "-alias",
+                "test-key",
+                "-keyalg",
+                "RSA",
+                "-keystore",
+                keyStorePath.toAbsolutePath().normalize().toString(),
+                "-storepass",
+                password,
+                "-keypass",
+                password,
+                "-dname",
+                "CN=localhost",
+                "-validity",
+                "365",
+                "-keysize",
+                "2048")
+            .inheritIO()
+            .start()
+            .waitFor();
+    if (exitCode != 0) {
+      throw new RuntimeException("Failed to generate keystore");
+    }
+    keyStore.load(Files.newInputStream(keyStorePath), password.toCharArray());
+    return keyStore;
+  }
+
+  public void start() {
+    try {
+      setupSSL();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    server.start();
+    sslServer.start();
+    isRunning = true;
+
+    System.out.println("HTTP server started at " + server.getAddress());
+    System.out.println("HTTPS server started at " + sslServer.getAddress());
+
+    try {
+      while (isRunning) {
+        Thread.sleep(500);
```

Comment:
Ideally, you'd block here on something like a semaphore, and release it from the `stop` method. Although for testing it should be fine.

Reply:
I guess you are right. It is code I inherited from the simple-httpbin project, and I just was moving it around, but I guess it may be worth improving this.